test/provider_os_memory_multiple_numa_nodes.cpp

```diff
@@ -75,14 +115,20 @@ struct testNumaNodes : public testing::TestWithParam<int> {
     umf_memory_provider_handle_t os_memory_provider = nullptr;
 };
 
+struct testNumaOnAllNodes : testNuma, testing::WithParamInterface<int> {};
+struct testNumaOnAllCpus : testNuma, testing::WithParamInterface<int> {};
+
 INSTANTIATE_TEST_SUITE_P(
-    testNumaNodesAllocations, testNumaNodes,
+    testNumaNodesAllocations, testNumaOnAllNodes,
     ::testing::ValuesIn(get_available_numa_nodes_numbers()));
 
+INSTANTIATE_TEST_SUITE_P(testNumaNodesAllocationsAllCpus, testNumaOnAllCpus,
+                         ::testing::ValuesIn(get_available_cpus()));
+
 // Test for allocations on numa nodes. This test will be executed for all numa nodes
 // available on the system. The available nodes are returned in vector from the
 // get_available_numa_nodes_numbers() function and passed to test as parameters.
-TEST_P(testNumaNodes, checkNumaNodesAllocations) {
+TEST_P(testNumaOnAllNodes, checkNumaNodesAllocations) {
```

Comment:
I think it would be good to have a test case for BIND where you set multiple NUMA nodes in node mask. This is allowed according to the documentation.

Reply:
I think we rely on the mbind behavior (proximity of the nodes) so it would be good to verify that.

Reply:
makes sense, you're right. Done.